tests/deployment_test.go

```diff
@@ -24,10 +24,269 @@
+		dynamicconfig.FrontendEnableWorkerVersioningDataAPIs.Key():     true,
+		dynamicconfig.FrontendEnableWorkerVersioningWorkflowAPIs.Key(): true,
+		dynamicconfig.FrontendEnableWorkerVersioningRuleAPIs.Key():     true,
+		dynamicconfig.FrontendEnableExecuteMultiOperation.Key():        true,
+		dynamicconfig.MatchingEnableDeployments.Key():                  true,
+		dynamicconfig.WorkerEnableDeployment.Key():                     true,
+	}
+	d.SetDynamicConfigOverrides(dynamicConfigOverrides)
+	d.FunctionalTestBase.SetupSuite("testdata/es_cluster.yaml")
+
+}
+
+func (d *DeploymentSuite) TearDownSuite() {
+	d.FunctionalTestBase.TearDownSuite()
+}
+
+func (d *DeploymentSuite) SetupTest() {
+	d.FunctionalTestBase.SetupTest()
+	d.setAssertions()
+}
+
+func (d *DeploymentSuite) TearDownTest() {
+}
+
+func (d *DeploymentSuite) TestDescribeDeployment_RegisterTaskQueue() {
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second*5)
+	defer cancel()
+
+	seriesName := "test"
+	buildID := "bgt"
+	taskQueue := &taskqueuepb.TaskQueue{Name: "deployment-test", Kind: enumspb.TASK_QUEUE_KIND_NORMAL}
+	workerDeployment := &deployment.Deployment{
+		SeriesName: seriesName,
+		BuildId:    buildID,
+	}
+	numberOfDeployments := 1
+
+	errChan := make(chan error)
+	defer close(errChan)
+
+	// Starting a deployment workflow
+	go func() {
+		_, err := d.FrontendClient().PollWorkflowTaskQueue(ctx, &workflowservice.PollWorkflowTaskQueueRequest{
+			Namespace: d.Namespace(),
+			TaskQueue: taskQueue,
+			Identity:  "random",
+			WorkerVersionCapabilities: &commonpb.WorkerVersionCapabilities{
+				UseVersioning:        true,
+				BuildId:              buildID,
+				DeploymentSeriesName: seriesName,
+			},
+		})
+		select {
+		case <-ctx.Done():
+			return
+		case errChan <- err:
+		}
+	}()
+
+	// Querying the Deployment
+	d.EventuallyWithT(func(t *assert.CollectT) {
+		a := assert.New(t)
+
+		resp, err := d.FrontendClient().DescribeDeployment(ctx, &workflowservice.DescribeDeploymentRequest{
+			Namespace:  d.Namespace(),
+			Deployment: workerDeployment,
+		})
+		a.NoError(err)
+		a.NotNil(resp.DeploymentInfo)
+		a.NotNil(resp.DeploymentInfo.Deployment)
+
+		a.Equal(seriesName, resp.DeploymentInfo.Deployment.SeriesName)
+		a.Equal(buildID, resp.DeploymentInfo.Deployment.BuildId)
+
+		if len(resp.DeploymentInfo.TaskQueueInfos) < numberOfDeployments {
+			return
+		}
+		a.Equal(numberOfDeployments, len(resp.DeploymentInfo.TaskQueueInfos))
+		a.Equal(taskQueue.Name, resp.DeploymentInfo.TaskQueueInfos[0].Name)
+		a.Equal(false, resp.DeploymentInfo.IsCurrent)
+	}, time.Second*5, time.Millisecond*200)
+
+	<-ctx.Done()
+	select {
+	case err := <-errChan:
+		d.Fail("Expected error channel to be empty but got error %w", err)
+	default:
+	}
+}
+
+func (d *DeploymentSuite) TestGetCurrentDeployment_NoCurrentDeployment() {
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second*5)
+	defer cancel()
+
+	seriesName := "test"
+	buildID := "bgt"
```

Comment:
Let's use `RandomizeStr` to make sure different tests do not interfere. (applicable to all tests)
```suggestion
	seriesName := RandomizeStr("my-series")
	buildID := RandomizeStr("bgt")
```